golang/cosmos/x/swingset/genesis.go

```diff
@@ -98,9 +98,8 @@ func ExportGenesis(ctx sdk.Context, k Keeper, swingStoreExportsHandler *SwingSto
 		// The export will fail if the export of a historical height was requested
 		snapshotHeight,
 		swingStoreGenesisEventHandler{exportDir: swingStoreExportDir, snapshotHeight: snapshotHeight},
-		// The export will fail if the swing-store does not contain all replay artifacts
 		keeper.SwingStoreExportOptions{
-			ArtifactMode:   keeper.SwingStoreArtifactModeReplay,
+			ArtifactMode:   keeper.SwingStoreArtifactModeOperational,
 			ExportDataMode: keeper.SwingStoreExportDataModeSkip,
```

Comment:
Hmm, shouldn't a genesis export require replay-level data?

Reply:
I don't think so, "operational" includes both XS heap snapshots and the subset (suffix) of transcript entries since the snapshot point.

I think a "genesis export" is meant to immediately reconstruct the chain state vector as of a given block height without retrieving or re-applying any cosmos txns. It could be used as the genesis of a new chain, with a different identity, whose initial state is equal to the most-recent state of the parent. And for that, you only need the current state of the vat workers, which is "operational" in our world.

In particular, it is *not* trying to start from the genesis state of the parent chain.

Now, the debugging/diagnostic replay-everything-with-a-different-xsnap things we might find ourselves needing to do *will* require replay-level data, and I suppose the question is whether we intend `agd genesis-export` to be the tool to fetch the data for that purpose, or if there's a different subcommand (or a `--replay` flag) that we'd use for that purpose.

Reply:
Right, at the end of the day, we can always devise a mechanism to re-import missing artifacts, as long as the hashes are correct in the export data. Genesis export does not change that. What may happen is that in a genesis export we'd need to edit artifacts in some fashion, which would require editing the hashes in the genesis file, but at this point I'm skeptical we'll ever be able to or need to do that.